internal/analytics/segment_reporter.go

```diff
@@ -80,6 +81,125 @@ func (r *SegmentReporter) RegisterCurrentIdentity(ctx context.Context, k8sCli ku
 	return nil
 }
 
+// ReportCommand reports a new executed command. The command should be anonymized before using this method.
+// The RegisterCurrentIdentity needs to be called first.
+func (r *SegmentReporter) ReportCommand(platform config.BotPlatform, command string) error {
+	return r.reportEvent("Command executed", map[string]interface{}{
+		"command":  command,
+		"platform": platform,
+	})
+}
+
+// ReportBotEnabled reports an enabled bot.
+// The RegisterCurrentIdentity needs to be called first.
+func (r *SegmentReporter) ReportBotEnabled(platform config.BotPlatform) error {
+	return r.reportEvent("Integration enabled", map[string]interface{}{
+		"platform": platform,
+		"type":     botIntegration,
+	})
+}
+
+// ReportSinkEnabled reports an enabled sink.
+// The RegisterCurrentIdentity needs to be called first.
+func (r *SegmentReporter) ReportSinkEnabled(platform Sink) error {
+	return r.reportEvent("Integration enabled", map[string]interface{}{
+		"platform": platform,
+		"type":     sinkIntegration,
+	})
+}
+
+// ReportBotHandledEventSuccess reports a successfully handled event using a given bot.
+// The RegisterCurrentIdentity needs to be called first.
+func (r *SegmentReporter) ReportBotHandledEventSuccess(platform config.BotPlatform) error {
```

Comment:
BTW, you also mentioned [batching](https://segment.com/docs/connections/sources/catalog/libraries/server/go/#batching) in the investigation docs, is it needed or not?

Reply:
But I also mentioned that it's built-in in the Segment Go client 🙂 https://segment.com/docs/connections/sources/catalog/libraries/server/go/

> - We should batch the events to avoid high resource consumption during heavy load. This could be a built-in feature, like in [Segment Go library](https://segment.com/docs/connections/sources/catalog/libraries/server/go/#batching).